src/components/LineGraph.vue

```diff
@@ -0,0 +1,382 @@
+<template>
+  <div class="col-12 h-100">
+    <div class="line-graph-hover row">
+      <div class="col text-start">
+        <h6 style="text-align: left; display: inline-block">
+          {{ name }}
+          <i :class="['fa-solid', icon]"></i>
+        </h6>
+      </div>
+      <div class="col text-end">
+        <span>
+          {{ scaleUnit(recordView[selectedField as keyof Record] as number)?.toFixed(2) ?? '0.00' }}
+          {{ unit }}&nbsp;
+        </span>
+      </div>
+    </div>
+    <div :ref="name"></div>
+  </div>
+</template>
+<script lang="ts">
+import { Record } from '@/spec/activity'
+import { Summary } from '@/spec/summary'
+import * as d3 from 'd3'
+
+export default {
+  props: {
+    name: {
+      type: String,
+      required: true
+    },
+    icon: String,
+    selectedField: { type: String, required: true },
```

Comment:
LineGraph intended for generate a Line Activity Graph, i advise `selectedField` rename or separate because it currently used for 2 purpose, selected field for record to be shown and calculate min/avg. Because min/avg have each formula for known fields, it's good to separate this OR add some information on it later.

Reply:
how about rename selectedField to "recordField"
then add 2 fields: summaryAvgField and summaryMaxField ?

Reply:
`recordField` looks good and `calculationField`? 
Or how about we move avg/max as function prop? so we can calculate independent each component call?

Reply:
ah you're right, let's make it props